src/process_utility.c

```diff
@@ -2568,8 +2568,34 @@ process_index_start(ProcessUtilityArgs *args)
 
 	if (NULL == ht)
 	{
-		ts_cache_release(hcache);
-		return DDL_CONTINUE;
+		/* Check if the relation is a Continuous Aggregate */
+		ContinuousAgg *cagg = ts_continuous_agg_find_by_rv(stmt->relation);
+
+		if (cagg)
+		{
+			/* If the relation is a CAgg and it is finalized */
+			if (ContinuousAggIsFinalized(cagg))
+				ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
+			else
+			{
+				ts_cache_release(hcache);
+				ereport(ERROR,
+						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						 errmsg("operation not supported on continuous aggreates that are not "
+								"finalized")));
+			}
+		}
+
+		if (NULL == ht)
+		{
+			ts_cache_release(hcache);
+			return DDL_CONTINUE;
+		}
+
+		/* Make the RangeVar for the underlying materialization hypertable */
+		stmt->relation = makeRangeVar((char *) NameStr(ht->fd.schema_name),
+									  (char *) NameStr(ht->fd.table_name),
+									  0);
```

Comment:
```suggestion
									  -1);
```
Shouldn't token location be unknown?

Reply:
Fixed!